tests/dummy/app/controllers/logging-example/application-log.js

```diff
@@ -0,0 +1,469 @@
+   @type String[]
+   */
+  _messageNumber: 0,
+
+  /**
+  Flexberry Logging Service
+
+   @property flexberryLoggingService
+   @type String[]
+   */
+  flexberryLoggingService: Ember.inject.service('flexberry-logging'),
+
+  /**
+  Current logging level
+
+  @property logLevel
+  @type Integer
+   */
+  logLevel: 0,
+
+  /**
+    List of available test application level settings: [ '0 OFF', 1: 'ERRORs', 2: 'ERRORs, WARNs', ... ].
+    First symbol - number of logging level, followed by list list of logging levels.
+
+    @property settings
+    @type String[]
+   */
+  settings: [],
+
+  /**
+   Default choise in settings
+
+   @property text
+   @type String
+   */
+  text: '',
+
+  /**
+   List buttons describers for Supported actions
+
+   @property customButtons
+   @type Object[]
+   */
+  customButtons:  Ember.computed('i18n.locale', function() {
+    let ret = [{
+      buttonName: this.get('i18n').t('forms.application-log.assert'),
+      buttonAction: 'assertAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: this.get('i18n').t('forms.application-log.error'),
+      buttonAction: 'errorAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: this.get('i18n').t('forms.application-log.throw'),
+      buttonAction: 'throwAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: this.get('i18n').t('forms.application-log.warn'),
+      buttonAction: 'warnAction',
+      buttonClasses: 'ui teal button'
+
+    }, {
+      buttonName: this.get('i18n').t('forms.application-log.log'),
+      buttonAction: 'logAction',
+      buttonClasses: 'ui teal button'
+    }, {
+      buttonName: this.get('i18n').t('forms.application-log.info'),
+      buttonAction: 'infoAction',
+      buttonClasses: 'ui teal button'
+    }, {
+      buttonName: this.get('i18n').t('forms.application-log.debug'),
+      buttonAction: 'debugAction',
+      buttonClasses: 'ui yellow button'
+
+    }, {
+      buttonName: this.get('i18n').t('forms.application-log.deprecation'),
+      buttonAction: 'deprecationAction',
+      buttonClasses: 'ui yellow button'
+    }];
+    return ret;
+  }),
+
+  _router: undefined,
+
+  /**
+  Selected jQuery object, containing HTML of confirm modal dialog.
+
+  @property confirmModalDialog
+  @type Object
+  @default undefined
+  */
+  confirmModalDialog: undefined,
+
+  /**
+  Content to be displayed in confirm modal dialog.
+  It will be displayed only if some confirm needs.
+
+  @property confirmModalDialogContent
+  @type String
+  @default ''
+   */
+  confirmModalDialogContent: '',
+
+  /**
+  Category to be passed in action confirm modal dialog: 'Error', 'Warn', 'Log', 'Info', ...
+
+  @property confirmModalDialogCategory
+  @type String
+  @default ''
+   */
+  confirmModalDialogCategory: '',
+
+  /**
+    Initialize properties
+
+    @method init
+   */
+  init() {
+    this.set('_router', getOwner(this).lookup('router:main'));
```

Comment:
Can't we use this.get('target').get('router') or used by you variant is better?.